src/openrct2/peep/Guest.cpp

```diff
@@ -4178,51 +4178,58 @@ void Guest::UpdateRideEnterVehicle()
 {
     Ride* ride = get_ride(current_ride);
 
-    rct_vehicle* vehicle = GET_VEHICLE(ride->vehicles[current_train]);
-    for (int32_t i = current_car; i != 0; --i)
+    if (ride != nullptr)
     {
-        vehicle = GET_VEHICLE(vehicle->next_vehicle_on_train);
-    }
+        rct_vehicle* vehicle = GET_VEHICLE(ride->vehicles[current_train]);
 
-    if (ride->mode != RIDE_MODE_FORWARD_ROTATION && ride->mode != RIDE_MODE_BACKWARD_ROTATION)
-    {
-        if (current_seat != vehicle->num_peeps)
-            return;
-    }
+        if (vehicle != nullptr)
+        {
+            for (int32_t i = current_car; i != 0; --i)
+            {
+                vehicle = GET_VEHICLE(vehicle->next_vehicle_on_train);
+            }
 
-    if (vehicle_is_used_in_pairs(vehicle))
-    {
-        auto seated_peep = (GET_PEEP(vehicle->peep[current_seat ^ 1]))->AsGuest();
-        if (seated_peep == nullptr || seated_peep->sub_state != PEEP_RIDE_ENTER_VEHICLE)
-            return;
+            if (ride->mode != RIDE_MODE_FORWARD_ROTATION && ride->mode != RIDE_MODE_BACKWARD_ROTATION)
+            {
+                if (current_seat != vehicle->num_peeps)
+                    return;
+            }
 
-        vehicle->num_peeps++;
-        ride->cur_num_customers++;
+            if (vehicle_is_used_in_pairs(vehicle))
+            {
+                auto seated_peep = (GET_PEEP(vehicle->peep[current_seat ^ 1]))->AsGuest();
```

Comment:
This hasn't actually fixed the source of the crash. This call to `AsGuest()` nullptr derefs if the paired peep doesn't exist. (This can happen when people really mess with the game)

Reply:
OK I'll take another look...